src/stores/right-panel/RightPanelStore.ts

```diff
@@ -0,0 +1,334 @@
+/*
+Copyright 2019 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { logger } from "matrix-js-sdk/src/logger";
+import { EventSubscription } from 'fbemitter';
+
+import defaultDispatcher from '../../dispatcher/dispatcher';
+import { pendingVerificationRequestForUser } from '../../verification';
+import SettingsStore from "../../settings/SettingsStore";
+import { RightPanelPhases/*, RIGHT_PANEL_PHASES_NO_ARGS */ } from "./RightPanelStorePhases";
+import { ActionPayload } from "../../dispatcher/payloads";
+import { Action } from '../../dispatcher/actions';
+import { SettingLevel } from "../../settings/SettingLevel";
+import { UPDATE_EVENT } from '../AsyncStore';
+import { ReadyWatchingStore } from '../ReadyWatchingStore';
+import RoomViewStore from '../RoomViewStore';
+import { IPhaseAndState, IPanelState, convertToStateRoom, convertToStoreRoom } from './RightPanelStoreIPanelState';
+
+const GROUP_PHASES = [
+    RightPanelPhases.GroupMemberList,
+    RightPanelPhases.GroupRoomList,
+    RightPanelPhases.GroupRoomInfo,
+    RightPanelPhases.GroupMemberInfo,
+];
+
+const MEMBER_INFO_PHASES = [
+    RightPanelPhases.RoomMemberInfo,
+    RightPanelPhases.Room3pidMemberInfo,
+    RightPanelPhases.EncryptionPanel,
+];
+
+/**
+ * A class for tracking the state of the right panel between layouts and
```

Comment:
This doesn't seem to be rewrapped to me...

Reply:
From @toger5, moved into the thread:

> This doesn't seem to be rewrapped to me...

I used this extension:https://marketplace.visualstudio.com/items?itemName=stkb.rewrap
And than added new lines where I though that they help with making the comment more structured.
Maybe I am misunderstanding what you mean with rewrap. Or should i just leave it as what I get from the extension?

Reply:
What I mean here is that the line lengths seem to differ throughout. For example, at least line 55 seems shorter than the rest. Starting new paragraphs is fine, but a single paragraph should have a consistent line line length of some kind. Also, if you put a line break inside the comment block between the paragraphs, Rewrap will keep them separated instead of jumbling them together.